main.nf

```diff
@@ -72,9 +72,22 @@ workflow {
     input:
         // Required
         input_gating_set_directory = Channel.fromPath(params.input_gating_set_directory)
-        active_channels_path = Channel.fromPath(params.active_channels_path)
-        channel_bounds_path = Channel.fromPath(params.channel_bounds_path)
-        supervised_list_path = Channel.fromPath(params.supervised_list_path)
+        // Optional - reasonable defaults if missing. 
+        if(params.active_channels_path){
+            active_channels_path = Channel.fromPath(params.active_channels_path)
+        }else{
+            active_channels_path = Channel.empty().ifEmpty("")
+        }
+        if(params.channel_bounds_path){
+            channel_bounds_path = Channel.fromPath(params.channel_bounds_path)
+        }else{
+            channel_bounds_path = Channel.empty().ifEmpty("")
+        }
+        if(params.supervised_list_path){
+            supervised_list_path = Channel.fromPath(params.supervised_list_path)
+        }else{
+            supervised_list_path = Channel.empty().ifEmpty("")
```

Comment:
`Channel.empty().ifEmpty("")` - https://www.nextflow.io/docs/latest/operator.html#operator-ifempty

Super super super cool.

This is basically a tool to set default channels in the case of a missing channel. I like it!